pkg/workflows/management/core_components.go

```diff
@@ -0,0 +1,110 @@
+package management
+
+import (
+	"context"
+
+	"github.com/aws/eks-anywhere/pkg/logger"
+	"github.com/aws/eks-anywhere/pkg/task"
+	"github.com/aws/eks-anywhere/pkg/types"
+	"github.com/aws/eks-anywhere/pkg/workflows"
+)
+
+type ensureEtcdCAPIComponentsExist struct{}
+
+// Run ensureEtcdCAPIComponentsExist ensures ETCD CAPI providers on the management cluster.
+func (s *ensureEtcdCAPIComponentsExist) Run(ctx context.Context, commandContext *task.CommandContext) task.Task {
+	logger.Info("Ensuring etcd CAPI providers exist on management cluster before upgrade")
+	if err := commandContext.CAPIManager.EnsureEtcdProvidersInstallation(ctx, commandContext.ManagementCluster, commandContext.Provider, commandContext.CurrentClusterSpec); err != nil {
+		commandContext.SetError(err)
+		return &workflows.CollectMgmtClusterDiagnosticsTask{}
+	}
```

Comment:
Why do we need a special check just for etcd controller? Why not check the other CAPI controllers?